src/eth/common.rs

```diff
@@ -0,0 +1,249 @@
+//! Common types encoding/decoding
+
+use lib::*;
+use super::{util, Stream, AbiType, Sink, Error};
+use parity_hash::Address;
+use bigint::U256;
+
+impl AbiType for u32 {
+	fn decode(stream: &mut Stream) -> Result<Self, Error> {
+		let previous_position = stream.advance(32);
+		if stream.position() > stream.payload().len() {
+			return Err(Error::UnexpectedEof);
```

Comment:
I'm wondering if these checks could be moved into `advance`?
So it would be 
```rust
stream.advance(32)?;
```

Reply:
Indeed